### ISSUE TITLE
implement proper lazy subscribers

### ISSUE THREAD

**Issue by mikeferguson** [ros-drivers/openni2_camera#119]
This is blocked by upstream work that should be included in Iron turtle - https://github.com/ros2/rmw/issues/330#issuecomment-1430800196

**Comment by mikeferguson** [comment]
Started on this - it is an epic rabbit hole:
 * We need to update image_proc/depth_proc to make this work (they aren't lazy, and they subscribe to our outputs)
 * That means we need to update image_transport (which most of the image_proc subscribers use)
 * That might also lead to changing things in message_filters, which image_transport uses (but so far, maybe not).

**Comment by mikeferguson** [comment]
Work in Progress:
 * https://github.com/ros-perception/image_common/pull/272
 * https://github.com/ros-perception/image_pipeline/pull/815
 * https://github.com/ros-drivers/openni2_camera/tree/proper_lazy_subs (will open a PR here once the image_common stuff is merged & released, since that is very much required)

**Comment by mikeferguson** [comment]
As a note - this will NOT make it into Iron, since we can't get the image_common changes merged into Iron (apparently that package is part of the base and was frozen on April 17). I will continue to get this merged into rolling, but it won't make into a regular release until 2024

**Comment by mikeferguson** [comment]
image_pipeline is now updated - but not yet released
